taroapp/cmd/ps.py

```diff
@@ -5,14 +5,21 @@
 from pygments.lexers.data import JsonLexer
 
 import taro.client
-from taro import dto
-from taro.jobs.job import JobInfoCollection
+from taro import dto, util
+from taro.jobs.job import JobInfoCollection,JobInfo
 from taroapp import printer
 from taroapp.view import instance as view_inst
 
 
 def run(args):
-    jobs = JobInfoCollection(*taro.client.read_jobs_info(args.instance))
+    if args.instance:
```

Comment:
The condition is not needed here.
The jobs as list can be filtered by list comprehension first and then added to new JobInfoCollection.